pkg/logql/engine.go

```diff
@@ -334,7 +334,13 @@ func (q *query) evalSample(ctx context.Context, expr syntax.SampleExpr) (promql_
 	}
 
 	if GetRangeType(q.params) == InstantType {
-		sort.Slice(vec, func(i, j int) bool { return labels.Compare(vec[i].Metric, vec[j].Metric) < 0 })
+		sortByValue, err := Sortable(q.params)
+		if err != nil {
+			return nil, err
+		}
+		if !sortByValue {
```

Comment:
I'm confused; why is this `!sortByValue`?

Reply:
`			sort.Slice(vec, func(i, j int) bool { return labels.Compare(vec[i].Metric, vec[j].Metric) < 0 })
`
Because this line of code exists before this PR, the meaning of this line of code seems to be` sortByLabel()`, but sort and sort_desc need `sortByValue()`, so when we have `sort` and `sort_desc` in our logql grammar, we should not execute `sortByLabel ()`

Reply:
The purpose of modifying this line of code is to make this test pass. 
I debugged for a long time at the beginning, and it was correctly sorted by value in the front, but the final output is indeed in the string order of the label.